stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/TypingIndicators.java

```diff
@@ -0,0 +1,44 @@
+package io.getstream.chat.docs.java;
+
+import io.getstream.chat.android.client.channel.ChannelClient;
+import io.getstream.chat.android.client.events.TypingStartEvent;
+import io.getstream.chat.android.client.events.TypingStopEvent;
+import kotlin.Unit;
+
+public class TypingIndicators {
+    private ChannelClient channelController;
+
+    /**
+     * @see <a href="https://getstream.io/chat/docs/typing_indicators/?language=java#sending-start-and-stop-typing-events">Sending Start and Stop Typing</a>
+     */
+    public void sendingStartAndStopTypingEvents() {
+        // Sends a typing.start event at most once every two seconds
+        channelController.keystroke().enqueue(result -> Unit.INSTANCE);
+
+        // Sends the typing.stop event
+        channelController.stopTyping().enqueue(result -> Unit.INSTANCE);
+    }
+
+    /**
+     * @see <a href="https://getstream.io/chat/docs/typing_indicators/?language=java#receiving-typing-indicator-events">Receiving Typing Events</a>
+     */
+    public void receivingTypingEvents() {
+        // Add typing start event handling
+        channelController.subscribeFor(
```

Comment:
poor Java programmers :-)

Reply:
We support Java, but we're letting them know that we don't actually want them to write Java code 😄